Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -334,17 +333,46 @@ void transportCovarianceToBound(
     const GeometryContext& geoContext, BoundSymMatrix& boundCovariance,
     BoundMatrix& fullTransportJacobian, FreeMatrix& freeTransportJacobian,
     FreeVector& freeToPathDerivatives, BoundToFreeMatrix& boundToFreeJacobian,
-    const FreeVector& freeParameters, const Surface& surface) {
+    FreeVector& freeParameters, const Surface& surface,
+    bool doFreeToBoundCorrection) {
   // Calculate the full jacobian from local parameters at the start surface to
   // current bound parameters
   boundToBoundJacobian(geoContext, freeParameters, boundToFreeJacobian,
                        freeTransportJacobian, freeToPathDerivatives,
                        fullTransportJacobian, surface);
 
-  // Apply the actual covariance transport to get covariance of the current
-  // bound parameters
-  boundCovariance = fullTransportJacobian * boundCovariance *
-                    fullTransportJacobian.transpose();
+  bool correction = false;
+  if (doFreeToBoundCorrection) {
+    BoundToFreeMatrix startBoundToFinalFreeJacobian =
+        freeTransportJacobian * boundToFreeJacobian;
+    FreeSymMatrix freeCovariance = startBoundToFinalFreeJacobian *
+                                   boundCovariance *
+                                   startBoundToFinalFreeJacobian.transpose();
+
+    auto transformer = detail::CorrectedFreeToBoundTransformer();
```

Comment:
I one would like to change the tuning parameters, this would be the place, right?

If that's so, we'd probably need to introduce a small struct 
```c++

struct FreeToBoundCorrection {

   bool apply = false;
  
 ActsScalar alpha = 0.1;
 
 ActsScalar beta = 2;

 ActsScalar& operator=(bool applyIt) {
  apply = applyIt;
   return (*this); 
}

 bool operator()() const {
   return apply;
}

};
```

Then this can be default instantiated in the method signatures with `const FreeToBoundCorrection& tfb = false` (hence we can drop the lang name because the struct type tells you what it is), but can be used to create a corrector as well, extracting the tunable parameters.

Reply:
To clarify, this struct could/should replace the `bool doFreeToBoundCorrecction` with a boolean to indicate if the correction is to be done, and with the parameters for the correction (in case the boolean is true).

Reply:
I see! Where should we define this struct?

Reply:
I think it should be defined either in a dedicated file, or in the `CorrectedFreeToBound` file.

Reply:
Thanks. I added a small struct `FreeToBoundCorrection` as suggested.